backend/src/modules/Boards/repository/mongo/MongoBoards.repository.js

```diff
@@ -30,6 +28,36 @@ export class MongoBoardsRepository {
     );
     return mongoDocumentToDomain(updatedBoard);
   }
+
+  async getBoardAnnouncements(boardID) {
+    const returnedData = await MongoBoardsModel.findById(boardID).select('announcements').exec();
+    const foundBoardAnnouncements = returnedData?.announcements?.map((announcement) =>
+      announcement.valueOf(),
+    );
+    return foundBoardAnnouncements;
+  }
+
+  async deleteBoardAnnouncement(announcementId) {
+    const foundBoardToUpdateAnnouncements = await MongoBoardsModel.findOne({
+      announcements: announcementId,
+    });
+    if (!foundBoardToUpdateAnnouncements) {
+      throw new Error('Board not found!');
+    }
+
+    const indexOfAnnouncementToDelete =
+      foundBoardToUpdateAnnouncements.announcements?.indexOf(announcementId);
+
+    foundBoardToUpdateAnnouncements?.announcements?.splice(indexOfAnnouncementToDelete, 1);
```

Comment:
po co Ci tutaj te znaki zapytania? 
if z 44 linii powinien wyeliminować to, że nie mamy boarda i pierwszy znak zapytania z 51 linii jest zbędny. A skoro każdy board ma announcements, to te znaki za announcements też są zbędne.

Reply:
Ok, faktycznie nadmiarowo